src/yyjson.h

```diff
@@ -799,6 +801,48 @@ yyjson_api_inline size_t yyjson_read_max_memory_usage(size_t len,
     return len * mul + pad;
 }
 
+yyjson_api bool unsafe_yyjson_read_number(const char *dat, size_t len,
+                                          void *val);
+
+/**
+ Read a JSON number.
+
+ This function is thread-safe when data is not modified by other threads.
+
+ @param dat The JSON data (UTF-8 without BOM), null-terminator is not required.
+    If this parameter is NULL, the function will fail and return false.
+ @param len The length of JSON data in bytes.
+    If this parameter is 0, the function will fail and return false.
```

Comment:
The null-terminator is required for the internal implementation of `read_number`.
The parameter `len` is not required.

Reply:
Yes, I've updated the comment message.
Not only the null-terminator is required, for example `123qwe` corretly parsed into `123` and `end` pointer is set to `q` character.